File: get_maps_images.py
import urllib.request
import re
import sys
import json

def get_images():
    query = "ABC Computer Training Centre Dabra Gwalior"
    url = f"https://www.google.com/search?q={urllib.parse.quote(query)}&tbm=isch"
    
    headers = {
        "User-Agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36"
    }
    
    req = urllib.request.Request(url, headers=headers)
    try:
        with urllib.request.urlopen(req) as response:
            html = response.read().decode('utf-8')
            
            # Find all googleusercontent.com URLs or standard image URLs in the search page
            images = re.findall(r'https://[a-zA-Z0-9.-]+\.googleusercontent\.com/[a-zA-Z0-9_/=-]+', html)
            images += re.findall(r'https://[a-zA-Z0-9.-]+\.ggpht\.com/[a-zA-Z0-9_/=-]+', html)
            images += re.findall(r'https://content\.jdmagicbox\.com/[a-zA-Z0-9_/=-]+\.jpg', html)
            
            # Remove duplicates while preserving order
            seen = set()
            unique_images = []
            for img in images:
                if img not in seen:
                    seen.add(img)
                    unique_images.append(img)
                    
            print(json.dumps(unique_images, indent=2))
            
    except Exception as e:
        print(f"Error fetching images: {e}", file=sys.stderr)

if __name__ == "__main__":
    get_images()

File: get_maps_images.js
const https = require('https');

function getImages() {
  const query = "ABC Computer Training Centre Dabra Gwalior";
  const url = `https://www.google.com/search?q=${encodeURIComponent(query)}&tbm=isch`;
  
  const options = {
    headers: {
      'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/91.0.4472.124 Safari/537.36'
    }
  };

  https.get(url, options, (res) => {
    let data = '';
    
    res.on('data', (chunk) => {
      data += chunk;
    });
    
    res.on('end', () => {
      const urls = [];
      
      // Match googleusercontent.com or ggpht.com or jdmagicbox image URLs
      const regex1 = /https:\/\/[a-zA-Z0-9.-]+\.googleusercontent\.com\/[a-zA-Z0-9_/=-]+/g;
      const regex2 = /https:\/\/[a-zA-Z0-9.-]+\.ggpht\.com\/[a-zA-Z0-9_/=-]+/g;
      const regex3 = /https:\/\/content\.jdmagicbox\.com\/[a-zA-Z0-9_/=-]+\.jpg/g;
      const regex4 = /https:\/\/[a-zA-Z0-9.-]+\.gstatic\.com\/images\?q=tbn:[a-zA-Z0-9_/=-]+/g;
      
      let match;
      while ((match = regex1.exec(data)) !== null) {
        urls.push(match[0]);
      }
      while ((match = regex2.exec(data)) !== null) {
        urls.push(match[0]);
      }
      while ((match = regex3.exec(data)) !== null) {
        urls.push(match[0]);
      }
      while ((match = regex4.exec(data)) !== null) {
        urls.push(match[0]);
      }
      
      const uniqueUrls = [...new Set(urls)];
      console.log(JSON.stringify(uniqueUrls, null, 2));
    });
    
  }).on('error', (err) => {
    console.error('Error fetching search page:', err.message);
  });
}

getImages();

